HW_3/Ex_18/Ex_18.py

```diff
@@ -0,0 +1,30 @@
+# Нужно создать рандомный массив A, который будет содержать определеннык кол-во цифр в любом порядке. Требуется найти в массиве A самый близкий по величине элемент к заданному числу X. Если есть число Х, то оно и будет самым близким.
+
+# _________________РЕШЕНИЕ__________________
+
+import random
+
+strt = int(input('Введите дипазон генерации.\n\nНачало диапазона: '))
+fin = int(input('Конец диапазона: '))
+
+size = int(input('Введите количество переменных: '))
+X = int(input('Какое число вы хотите обнаружить в массиве?\nВведите: '))
+
+A = [random.randint(strt, fin) for i in range(size)]
+print(f'\nРандомный массив:\n{A}')
+
+if A[0] <= X: min_diff = X - A[0]
+else: min_diff = A[0] - X
```

Comment:
Такая запись хоть и допустима, но не рекомендуется.
```suggestion
if A[0] <= X:
    min_diff = X - A[0]
else:
    min_diff = A[0] - X
```

Reply:
Обратите внимание на функцию `abs`. Она тут поможет

Reply:
Пытался через эту функцию. Не смог придумать как её тут задействовать.

Reply:
Функция abc возвращает абсолютное значение, поэтому использовать ее очень просто
```suggestion
min_diff = abs(A[0] - X)
```

Reply:
Извиняюсь за столь очевидный затуп. Я действительно пытался, но писал все время `A[0] - abs(X)`. Спасибо за подсказку)